dhctl/pkg/preflight/common.go

```diff
@@ -49,7 +49,7 @@ func setupSSHTunnelToProxyAddr(sshCl node.SSHClient, proxyUrl *url.URL) (node.Tu
 		}
 	}
 	var tunnel string
-	if app.SSHLegacyMode {
+	if app.SSHLegacyMode || (len(app.SSHPrivateKeys) > 0 && !app.SSHModernMode) {
```

Comment:
Please move this condition to function in `sshclient/init.go`

For example with name `NeedToUseLegacyClient`